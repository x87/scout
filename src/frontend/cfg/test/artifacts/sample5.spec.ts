import loop5 from 'samples/loop5';
import { decompile } from '../util';

`
{$CLEO}

// this demonstrates a problem from CAT1 script

0000:

jump @middle


while 0@ > 1

    wait 1
    
    :middle
    wait 2


end

end_thread
`;

describe(`loop5`, () => {
  it(`does not detect loop correctly as there is a jump in the middle of the loop`, async () => {
    expect(await decompile(loop5)).toMatchSnapshot();
  });
});
