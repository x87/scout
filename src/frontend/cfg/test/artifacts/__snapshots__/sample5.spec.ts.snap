// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loop5 does not detect loop correctly as there is a jump in the middle of the loop 1`] = `
"
:NONAME_0
    NOP
    
    while true
        WAIT 2
        if 
            IS_INT_LVAR_GREATER_THAN_NUMBER 0 1
        then
        end
        WAIT 1
    end
    
    TERMINATE_THIS_SCRIPT
"
`;
